### ISSUE TITLE
How to keep Login session for long time in iOS

### ISSUE THREAD

**Issue by anil03** [QuickBlox/quickblox-ios-sdk#325]
hi Team,

We are using the QB account API for the our application. We need 
1. We want to remove the listing page for calling purpose. we want to select he user internally and cal him directly.
2. We want to have longer time period for Login Session for user. How can we achieve this functionality into ios.

**Comment by Raikerian** [comment]
Hi,

Please specify your questions in a little bit more details:

1. Not sure what do you mean here. Are you talking about audio/video calling?
2. Our session is expiring after 2 hours by default, we can make it longer, but only if you are having our dedicated instance (e.g. Enterprise account). There is no need to handle this sessionm in a latest versions of our SDK. Every QBRequest method will first check if session is valid, if not - it will renew it and then perform an operation you were calling for.

**Comment by anil03** [comment]
I am making an application which requires video chat as main functionality,for which i am using quickblox sdk.After referring the demo and implementing it everything works just fine,but i need to sign in each time i am making the video call.Can't we maintain session which requires the user to sign in only once.In this tutorial https://github.com/QuickBlox/quickblox-ios-sdk/tree/master/sample-videochat-webrtc user can receive video call only if he is screen no 2 i,e screen which appears after login.Can't we maintain the login status once the user has logged in and until he logs out ?

As in where the login session is destoyed in the given above demo?

**Comment by iSevenDays** [comment]
Hi anil03,

In order to call opponent user(recipient) you should only login to QBChat.

Then you should send presence message to a server every 30-45 seconds.

http://quickblox.com/developers/SimpleSample-chat_users-ios#Keep_alive_chat_connection_with_presence

**Comment by anil03** [comment]
Can't i make multiple video call by logging only once?can't i maintain login session of the user throughout the application?

**Comment by iSevenDays** [comment]
anil03, you can make as many calls as you want if [[QBChat instance] isConnected]

Please refer to the documentation for Chat http://quickblox.com/developers/SimpleSample-chat_users-ios

**Comment by anil03** [comment]
but my login session is not maintained it gets destroyed and i am not able to receive videocall 

- (void)didReceiveNewSession:(QBRTCSession *)session userInfo:(NSDictionary *)userInfo 
this delegate is never called on the opposite side

**Comment by iSevenDays** [comment]
QBChat disconnects without presence messages

In order to call opponent user(recipient) you should be logged in to QBChat.

Then you should send presence message to a server every 30-45 seconds.

Please run our sample and try to reproduce your issue.

If didReceiveNewSession is not called on the opposite side, please ensure that you added your class to delegates of QBRTCClient

Take a look at our sample video chat and try to reproduce the issue

https://github.com/QuickBlox/quickblox-ios-sdk/tree/master/sample-videochat-webrtc

**Comment by anil03** [comment]
(void)didReceiveNewSession:(QBRTCSession *)session userInfo:(NSDictionary *)userInfo this delegate is called only once after the login.Once i had successfully made  video call,the next time this delegate is not called.

what does this mean Then you should send presence message to a server every 30-45 seconds.?

**Comment by iSevenDays** [comment]
anil03, please see my answer above https://github.com/QuickBlox/quickblox-ios-sdk/issues/325#issuecomment-194256172

**Comment by anil03** [comment]
I just want incomingviewcontroller to appear anywhere in the application if the opponent is calling,for that i have to maintain session.

**Comment by Raikerian** [comment]
You just need to automatically send presence every 30-45 seconds, then your session will be alive :)

**Comment by Raikerian** [comment]
Were you able to accomplish it?

**Comment by Raikerian** [comment]
If you are still having any other questions feel free to let us know by creating a separate issue.
